Did some reorganization of the notebook and added a code chunk to do webcrawling of BM.

MKLau opened the pull request:
I did some code consultation with ChatGPT and Gemini. Take a look at my code and let me know what you think. 

Don't approve the pull request, just wanted to have folks take a look. 

*Heads-up: I don't recommend running unless you have a stable connection.*

JeffBaucom commented:
@MKLau what is the expected behavior of the last chunk? I think my R packages might be incorrectly installed, it seemed to fail at the depth-2 scraping because the final table I received is the same as the depth1 progress table you committed.

MKLau commented:
I didn't finish running on my  machine, but it was scraping the depth 2 pages. That output should be there, if it's not, we should check the loop that's running the depth 2 scrape.

MKLau commented:
Yeah, per your comment @JeffBaucom, there were some bugs introduced in the loading of package dependencies. I fixed those bugs and cleaned up the loading of libraries, so you shouldn't have any dep issues. 

Per expected behavior, it should produce both a table scraped from the database table listing all of the species by Hawaiian name and lower down in the Webcrawl subsection, it should produce another table with the scraped pages at depth 2. But, I'm running the code now to check.

MKLau commented:
Just finished re-running and  you were correct. This is crawling all sub-pages to get the full list of species in the database. We'll need to now add the depth 2 scraping to this list of URLs.

MKLau commented:
Notebook should now install all needed dependencies and render. You'll need to install `rmarkdown` manually first and then run `render('notebook.Rmd')`. 

@stacyllyn @JeffBaucom

JeffBaucom commented on notebook.Rmd:
```diff
@@ -69,16 +64,370 @@ write.csv(ethno_table, file = "data/ethno_table.csv", row.names = FALSE)
+new_data <- depth1_dt[Detail_URL %in% pending_urls]
+final_batch <- cbind(new_data, details_dt)
+
+                                        # Append to final dataset
```
@MKLau  I'm seeing some indentation on these comments, is it possible to fix those or are they indented like that on purpose?

MKLau replied:
Happy to fix that and any other formatting issues. That's just what was returned by OpenAI. I didn't spend a lot of time on it because I went through multiple iterations.
